[WFLY-7205] Transaction timeout tests

ochaloup opened the pull request:
Adding testcases aiming to test transaction timeout of ejb transactions.

Having found jiras
https://issues.jboss.org/browse/WFLY-6212
https://issues.jboss.org/browse/WFLY-6585

EAP7 PR: jbossas/jboss-eap7#783

ochaloup committed:
[WFLY-4672] Adding tests for exception being thrown from SFSB

ochaloup committed:
[WFLY-7205] Adding tests for transaction timeout behavior

[WFLY-7205] Adding tests for transaction timeout behavior when BMT is used
[WFLY-7205] Adding tests for transaction timeout behavior for MDB

ochaloup committed:
[WFLY-7205][WFLY-6585] Adding tests for transaction timeout behavior for IIOP and EJB2 bean

ochaloup committed:
[WFLY-7205][refactoring] removing unsused imports from transaction testing files

[WFLY-7205][refactoring] Removing meaningless setup task for JTS

There is no worthy to have the task defined or used as JTS for being set
needs restart of server which is not done either here or even under
basic set of tests.
Setting of JTS and orb is done by XSLT template when jboss as is copied
to target directory.

ochaloup committed:
[WFLY-7205] Adding newpackage with utility classes for transactions under shared module of testsuite

tadamski committed:
IIOPTimeoutTestCase change expected exception

ochaloup commented:
EAP7 PR: https://github.com/jbossas/jboss-eap7/pull/783

kabir commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/bmt/timeout/StatelessBmtRestartTimeoutBean.java:
```diff
@@ -0,0 +1,73 @@
+        }
+
+        // reseting transaction timeout to default one
+        tm.setTransactionTimeout(0);
```
I think this needs to happen in a finally block

kabir commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/bmt/timeout/StatelessBmtBean.java:
```diff
@@ -0,0 +1,116 @@
+        TxTestUtil.checkTransactionExists(tm, false);
+
+        for(int i = 1; i <= transactionCount; i++) {
+            boolean isTimeout = (timeoutCount - i >= 0) ? true : false;
```
Why not simply?
    (timeoutCount - i >= 0)
or simpler:
    i <= timeoutCount

And in a simple test ver see this becoming

ochaloup replied:
yes, I don't know why I wrote it that complicated way :/

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulTimeoutTestCase.java:
```diff
@@ -0,0 +1,344 @@
+ * if transaction is aborted.
+ */
+@RunWith(Arquillian.class)
+@Ignore("WFLY-6212")
```
Do not add new tests that are @ignored at there inception.
once WFLY-6212 is fixed, please submit this tests with it as well. 
but do not add new test that dead on arrival.

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationAndInterfaceBean.java:
```diff
@@ -0,0 +1,99 @@
+    private TransactionCheckerSingleton checker;
+
+    public void afterBegin() throws EJBException, RemoteException {
+        log.info("afterBegin called");
```
why? 
log.trace if anything at all.

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationAndInterfaceBean.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    public void beforeCompletion() throws EJBException, RemoteException {
+        log.info("beforeCompletion called");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationAndInterfaceBean.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    public void afterCompletion(boolean committed) throws EJBException, RemoteException {
+        log.infof("afterCompletion: transaction was%s committed", committed ? "" : " not");
```
log.tracef if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationAndInterfaceBean.java:
```diff
@@ -0,0 +1,99 @@
+
+    @AfterBegin
+    public void afterB() {
+        log.info("afterBegin called");
```
log.trace if anythinglog.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationAndInterfaceBean.java:
```diff
@@ -0,0 +1,99 @@
+
+    @BeforeCompletion
+    public void beforeC() {
+        log.info("beforeCompletion called");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationAndInterfaceBean.java:
```diff
@@ -0,0 +1,99 @@
+
+    @AfterCompletion
+    public void afterC(boolean isCommitted) {
+        log.infof("afterCompletion: transaction was%s committed", isCommitted ? "" : " not");
```
log.tracef if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationBean.java:
```diff
@@ -0,0 +1,114 @@
+
+    @AfterBegin
+    public void afterBegin() {
+        log.info("afterBegin called");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationBean.java:
```diff
@@ -0,0 +1,114 @@
+
+    @BeforeCompletion
+    public void beforeCompletion() {
+        log.info("beforeCompletion called");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationBean.java:
```diff
@@ -0,0 +1,114 @@
+
+    @AfterCompletion
+    public void afterCompletion(boolean isCommitted) {
+        log.infof("afterCompletion: transaction was%s committed", isCommitted ? "" : " not");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithAnnotationBean.java:
```diff
@@ -0,0 +1,114 @@
+    }
+
+    public void touch() {
+        log.info("Bean touched");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatefulWithRegistryBean.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    public void touch() {
+        log.info("Bean touched");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatelessWithAnnotationBean.java:
```diff
@@ -0,0 +1,79 @@
+
+    @AfterBegin
+    public void afterBegin() {
+        log.info("afterBegin called");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatelessWithAnnotationBean.java:
```diff
@@ -0,0 +1,79 @@
+
+    @BeforeCompletion
+    public void beforeCompletion() {
+        log.info("beforeCompletion called");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/cmt/timeout/xa/StatelessWithAnnotationBean.java:
```diff
@@ -0,0 +1,79 @@
+
+    @AfterCompletion
+    public void afterCompletion(boolean isCommitted) {
+        log.infof("afterCompletion: transaction was%s committed", isCommitted ? "" : " not");
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/ActivationPropertyTimeoutMDB.java:
```diff
@@ -0,0 +1,92 @@
+    @Override
+    public void onMessage(Message message) {
+        try {
+            log.infof("onMessage received message: %s '%s'", message, ((TextMessage) message).getText());
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/ActivationPropertyTimeoutMDB.java:
```diff
@@ -0,0 +1,92 @@
+
+            // this should mean that second attempt for calling onMessage comes to play
+            if(checker.getRolledback() > 0) {
+                log.infof("Discarding message '%s' as onMessage called for second time", message);
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/AnnotationTimeoutMDB.java:
```diff
@@ -0,0 +1,90 @@
+    @Override
+    public void onMessage(Message message) {
+        try {
+            log.infof("onMessage received message: %s '%s'", message, ((TextMessage) message).getText());
```
log.trace if anything

ctomc commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/NoTimeoutMDB.java:
```diff
@@ -0,0 +1,82 @@
+    @Override
+    public void onMessage(Message message) {
+        try {
+            log.infof("onMessage received message: %s '%s'", message, ((TextMessage) message).getText());
```
log.tracef

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/IIOPTimeoutTestCase.java:
```diff
@@ -0,0 +1,184 @@
+    }
+
+    @Test
+    @Ignore("WFLY-6935")
```
please do not add ignore tests. 
Add them once jira is fixed.

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatefulBean.java:
```diff
@@ -0,0 +1,102 @@
+    private TransactionCheckerSingleton checker;
+
+    public void afterBegin() throws EJBException, RemoteException {
+        log.info("afterBegin called");
```
log.trace

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatefulBean.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public void beforeCompletion() throws EJBException, RemoteException {
+        log.info("beforeCompletion called");
```
log.trace

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatefulBean.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public void afterCompletion(boolean committed) throws EJBException, RemoteException {
+        log.infof("afterCompletion: transaction was%s committed", committed ? "" : " not");
```
log.tracef

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatefulBean.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public void ejbCreate() throws java.rmi.RemoteException, javax.ejb.CreateException {
+        log.debugf("Creating method for home interface '%s' was invoked", this.getClass());
```
log.tracef

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatefulBean.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public void testTransaction() throws RemoteException, SystemException {
+        log.info("Method stateful #testTransaction called");
```
log.tracef

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatefulBean.java:
```diff
@@ -0,0 +1,102 @@
+
+    @TransactionTimeout(value = 1)
+    public void testTimeout() throws SystemException, RemoteException {
+        log.info("Method stateful #testTimeout called");
```
log.tracef

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatefulBean.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    public void touch() {
+        log.debug("Stateful bean instance has been touched");
```
log.trace

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatelessBean.java:
```diff
@@ -0,0 +1,97 @@
+    private TransactionCheckerSingleton checker;
+
+    public void afterBegin() throws EJBException, RemoteException {
+        log.info("afterBegin called");
```
log.trace if at all

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatelessBean.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    public void beforeCompletion() throws EJBException, RemoteException {
+        log.info("beforeCompletion called");
```
log.trace if at all

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatelessBean.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    public void afterCompletion(boolean committed) throws EJBException, RemoteException {
+        log.infof("afterCompletion: transaction was%s committed", committed ? "" : " not");
```
log.tracef

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatelessBean.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    public void testTransaction() throws RemoteException, SystemException {
+        log.info("Method stateless #testTransaction called");
```
log.trace if at all

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatelessBean.java:
```diff
@@ -0,0 +1,97 @@
+
+    @TransactionTimeout(value = 1)
+    public void testTimeout() throws SystemException, RemoteException {
+        log.info("Method stateless #testTimeout called");
```
log.trace if at all

ctomc commented on testsuite/integration/iiop/src/test/java/org/jboss/as/test/iiop/transaction/timeout/StatelessBean.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    public void touch() {
+        log.debug("Stateless bean instance has been touched");
```
log.trace

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/ManagementUtil.java:
```diff
@@ -0,0 +1,96 @@
+ *
+ * @author Ondra Chaloupka <ochaloup@redhat.com>
+ */
+public final class ManagementUtil {
```
Another?
Is it enough we have at least 5 others yet. 
With pretty much same method names and what they do.

kabir replied:
We already have ManagementOperations in WFCORE's testsuite/shared, so I think we should use that. If it doesn't meet all your needs that class should be extended.

ochaloup replied:
sure, please, can you give me a clue what is the right one to be used?
I didn't want to add a new one but I wasn't been able to find some generic one for me to use.

ochaloup replied:
sure, thanks for pointing me to the wfcore

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestSynchronization.java:
```diff
@@ -0,0 +1,58 @@
+
+    @Override
+    public void beforeCompletion() {
+        log.infof("beforeCompletion called");
```
log.trace if at all

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestSynchronization.java:
```diff
@@ -0,0 +1,58 @@
+     */
+    @Override
+    public void afterCompletion(int status) {
+        log.infof("afterCompletion called with status '%s'", status);
```
log.trace if at all

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public void commit(Xid xid, boolean onePhase) throws XAException {
+        log.infof("commit xid:[%s], %s one phase", xid, onePhase ? "with" : "without");
```
log.trace

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public void end(Xid xid, int flags) throws XAException {
+        log.infof("end xid:[%s], flag: %s", xid, flags);
```
trace

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public void forget(Xid xid) throws XAException {
+        log.infof("forget xid:[%s]", xid);
```
trace

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public int getTransactionTimeout() throws XAException {
+        log.infof("getTransactionTimeout: returning timeout: %s", transactionTimeout);
```
tracef

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public boolean isSameRM(XAResource xares) throws XAException {
+        log.infof("isSameRM returning false to xares: %s", xares);
```
tracef

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public int prepare(Xid xid) throws XAException {
+        log.infof("prepare xid: [%s]", xid);
```
tracef

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public Xid[] recover(int flag) throws XAException {
+        log.infof("recover with flags: %s", flag);
```
tracef

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public void rollback(Xid xid) throws XAException {
+        log.infof("rollback xid: [%s]", xid);
```
tracef

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public boolean setTransactionTimeout(int seconds) throws XAException {
+        log.infof("setTransactionTimeout: setting timeout: %s", seconds);
```
tracef

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TestXAResource.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public void start(Xid xid, int flags) throws XAException {
+        log.infof("start xid: [%s], flags: %s", xid, flags);
```
tracef

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TransactionTestLookupUtil.java:
```diff
@@ -0,0 +1,108 @@
+
+    public static <T> T lookupModule(InitialContext ctx, Class<T> beanType) throws NamingException {
+        String lookupString = String.format("java:module/%s!%s", beanType.getSimpleName(), beanType.getName());
+        log.debug("looking for: " + lookupString);
```
trace

ctomc commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TransactionTestLookupUtil.java:
```diff
@@ -0,0 +1,108 @@
+            Class<T> remoteInterface, boolean isStateful) throws NamingException {
+        String ejbLookup = org.jboss.as.test.shared.integration.ejb.security.Util.createRemoteEjbJndiContext(
+                "", archiveName, "", beanType.getSimpleName(), remoteInterface.getName(), isStateful);
+        log.debug("looking for: " + ejbLookup);
```
trace

kabir commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/ManagementUtil.java:
```diff
@@ -0,0 +1,96 @@
+    private static final Logger log = Logger.getLogger(ManagementUtil.class);
+
+    public static ModelNode readAttribute(ModelNode address, String name, ManagementClient mgmtClient) throws IOException, MgmtOperationException {
+        ModelNode operation = new ModelNode();
```
This can already be done with 
`import org.jboss.as.controller.operations.common.Util;
...
        ModelNode op = Util.getReadAttributeOperation(PathAddress.pathAddress(address), name);
`
include-defaults defaults to true anyway

ochaloup replied:
ok ,thanks

stuartwdouglas commented:
Can these tests have some comments added to them so that is is obvious exactly what they are testing?

ochaloup commented:
@stuartwdouglas: sure, I will add comments with description

ochaloup commented:
@ctomc @kabir @stuartwdouglas : I tried to address all your objections. Can you please review this once again? Thanks in advance.

note: once the PR pass the review I will squash multiple commits to one.

ochaloup commented:
My wrong, tested before rebase against master was done. Hopefully now the troubles of ci are fixes.

ochaloup commented:
I didn't realize that PR#9112 was not merged and tests could fail.

ochaloup commented:
@kabir as Tomek's fix on IIOP was merged (wildfly/wildfly#9112) I return back the test under IIOP timeout testcase and squashed changes based on comments in this PR. I hope that I addressed all of them. At that way this PR would be ready for final review(@stuartwdouglas?)/merge

I'm not sure why the test `WSTrustTestCase` fails but I think it's not connected to this PR.

kabir commented:
Retest this please

kabir commented:
@stuartwdouglas Please approve

stuartwdouglas commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/bmt/timeout/StatelessTimeoutTestCase.java:
```diff
@@ -0,0 +1,123 @@
+     */
+    @Test
+    public void timeout() throws Exception {
+        StatelessBmtBean bean = TransactionTestLookupUtil.lookupModule(initCtx, StatelessBmtBean.class);
```
These tests seem like a fundamental test of the transaction manager, and don't appear to be EJB specific. I don't really know if these are needed, as I assume this would be tested fairly comprehensively by the Narayana test suite.

@tomjenkinson do you think it is worth having these tests? They basically just test that a transaction than times out is not committed.

stuartwdouglas commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/mdb/timeout/MessageDrivenTimeoutTestCase.java:
```diff
@@ -0,0 +1,176 @@
+    @Test
+    public void noTimeout() throws Exception {
+        String text = "no timeout";
+        Queue q = sendMessage(text, TransactionTimeoutQueueSetupTask.NO_TIMEOUT_JNDI_NAME);
```
Once again this just seems like a test of basic transaction functionality?

tomjenkinson commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/exception/ExceptionTestCase.java:
```diff
@@ -43,47 +45,109 @@
+        SFSB1Interface sfsb1 = getBean();
+        Assert.assertFalse(isPreDestroy.is());
+        try {
+            sfsb1.userException();
```
I think you need a fail below this

tomjenkinson commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/stateful/exception/ExceptionTestCase.java:
```diff
@@ -43,47 +45,109 @@
+        Assert.assertFalse(isPreDestroy.is());
+
+        try {
+            sfsb1.userException();
```
fail below this?

ochaloup replied:
fixed

tomjenkinson commented on testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/bmt/timeout/StatelessBmtBean.java:
```diff
@@ -0,0 +1,116 @@
+        try {
+            return TxTestUtil.getStatusAsString(txn.getStatus());
+        } catch (SystemException se) {
+            throw new EJBException("Can't get status of transaction", se);
```
Why not allow the SystemException up?

ochaloup replied:
True, I can's see any reason why it should not be thrown. Can't remember my reason for this.

tomjenkinson commented on testsuite/shared/src/main/java/org/jboss/as/test/integration/transactions/TxTestUtil.java:
```diff
@@ -0,0 +1,123 @@
+            if(!isExpectTransaction && tx != null && tx.getStatus() != Status.STATUS_NO_TRANSACTION) {
+                throw new RuntimeException("We do not expect transaction would be active - we haven't activated it in BMT bean");
+            } else if (isExpectTransaction && (tx == null || tx.getStatus() != Status.STATUS_ACTIVE)) {
+                throw new RuntimeException("We do expect tranaction would be active - we have alredy activated it in BMT bean");
```
maybe just fail("MESSAGE")

ochaloup replied:
ah, right, absolutely agree

tomjenkinson commented:
I can see @stuartwdouglas point, @ochaloup perhaps you can explain why these integration tests are best placed in WFLY rather than unit tests in respective projects? Perhaps the need to be here is something to do with ensuring something in the EJB interceptors is working as expected (rolling back TX, garbage collection?)?

On the face of it the test goes:
begin()
enlist x2
wait for timeout
verify commit works/not

With more details on why this testing is most appropriate in WFLY it will be easier to review.

For example, what is the purpose of introducting the MDB rather than an additional Test XAR? That might help explain why the test needs to be here.

Thanks!

ochaloup commented:
Hi Stuart, Hi Tom,

tests are thought to check integration of txn processing in app server.

That's why I chose them having in WFLY testsuite (and in fact these tests uncover two issues here). It checks app server and txn processing spec compliance (at least a little bit) - here from specific point of view of exception handling on txn processing and txn timeout happening.

I agree with Stuart that the 'noTimeout' testcases are not relevant from this point of view. But we (QE) run wfly testsuite as a kind of initial check if everything works fine.
E.g. when checking some older versions with minor updates is the base of our testing too. We do not have enough resources to run all component testsuites at such cases. That's reason why I proposed that simple (and probably duplication) tests to WFLY testsuite. If it does not comply with rules for testsuite I can remove them.

I hope this clarify what's the idea behind here
Thank you

tomjenkinson commented:
Another observation is that if the transaction manager was ever replaced with an alternative transaction manager it can add serve as protection to WFLY to test this in integration suite.

ochaloup commented:
I squashed the changes and updated the eap7 PR to be in sync with this one
